apps/trading/e2e/tests/referrals/test_referrals.py

```diff
@@ -0,0 +1,131 @@
+
+def check_volume_and_tier(page: Page, expected_volume, expected_tier):
+    if "referrals" in page.url:
+         page.reload()
+    else:
+         page.goto("/#/referrals/")
+
+    combined_volume_element = page.get_by_test_id('combined-volume-value')
+    current_tier_element = page.get_by_test_id('current-tier-value')
+
+    volume_matched = element_contains_text(combined_volume_element, expected_volume)
+    tier_matched = element_contains_text(current_tier_element, expected_tier)
+
+    return volume_matched, tier_matched
+
+@pytest.mark.usefixtures("page", "auth", "risk_accepted")
+def test_referral_scenario(continuous_market, vega: VegaService, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_A)
+
+    create_and_faucet_wallet(vega=vega, wallet=PARTY_B)
+    vega.wait_for_total_catchup()
+    vega.update_referral_program(
+        proposal_key=MM_WALLET.name,
+        benefit_tiers=[
+            {
+                "minimum_running_notional_taker_volume": 100,
+                "minimum_epochs": 1,
+                "referral_reward_factor": 0.01,
+                "referral_discount_factor": 0.01,
+            },
+            {
+                "minimum_running_notional_taker_volume": 200,
+                "minimum_epochs": 2,
+                "referral_reward_factor": 0.02,
+                "referral_discount_factor": 0.02,
+            },
+            {
+                "minimum_running_notional_taker_volume": 300,
+                "minimum_epochs": 3,
+                "referral_reward_factor": 0.03,
+                "referral_discount_factor": 0.03,
+            },
+        ],
+        staking_tiers=[
+            {"minimum_staked_tokens": 100, "referral_reward_multiplier": 1},
+            {"minimum_staked_tokens": 1000, "referral_reward_multiplier": 2},
+            {"minimum_staked_tokens": 10000, "referral_reward_multiplier": 2},
+        ],
+        window_length=1,
+    )
+    next_epoch(vega=vega)
+    vega.wait_for_total_catchup()
+    vega.create_referral_set(key_name=PARTY_A.name)
+    vega.wait_fn(10)
+    vega.wait_for_total_catchup()
+    print(vega.list_referral_sets())
+    referral_set_id = list(vega.list_referral_sets().keys())[0]
+    print(referral_set_id)
+    vega.apply_referral_code(key_name=PARTY_B.name, id=referral_set_id)
+
+    tdai_id = vega.find_asset_id(symbol="tDAI")
+
+    vega.mint(
+        "Key 1",
+        asset=tdai_id,
+        amount=10e6,
+    )
+
+    vega.mint(
+        PARTY_B.name,
+        asset=tdai_id,
+        amount=10e6,
+    )
+    submit_liquidity(vega, MM_WALLET.name, continuous_market, 100, 100)
+
+    vega.wait_fn(1)
+    vega.wait_for_total_catchup()
+
+    change_keys(page, vega, PARTY_B.name)
+    submit_order(vega, PARTY_B.name, continuous_market, "SIDE_BUY", 1, 115)
+    vega.wait_for_total_catchup()
+    next_epoch(vega=vega)
+    volume_matched, tier_matched = check_volume_and_tier(page, r'^1\d{2}$', "3")
```

Comment:
We shouldn't need regex because we know the exact volume?
also in the other checks you are creating a pattern but you haven't in this one.

Reply:
Couple reasons for this.
1. I do not totally trust doing this because there were magic values buried everywhere
2. I am pretty sure the default is that the values are pegged with the liquidity submitted

Long story short with those above points is that I would want to refactor this so I can be sure of my own values because right now it feels brittle to me. I would also hope that the unit tests check the calculations. So this _shouldn't_ be a gap. I just wanted to reflect moving through the tiers... let me know if you do not agree with that logic though